crates/runtime/src/component/blobstore.rs

```diff
@@ -99,7 +99,14 @@ where
             (Ok(stream), io) => {
                 if let Some(io) = io {
                     // TODO: Move this into the runtime
-                    io.await.context("failed to perform async I/O")?;
+                    let handle = tokio::spawn(async move {
+                        if let Err(e) = io.await {
+                            tracing::error!("Error awaiting async IO: {:?}", e);
+                        }
+                    });
+                    self.table
+                        .push(handle)
+                        .context("failed to push async I/O")?;
```

Comment:
Looks like this would make consuming data non-deterministic, reading: nothing, complete buffer or anywhere in between. In other words, this would introduce a race condition between `io.await` being polled until it's ready by Tokio runtime and the component consuming the value (https://github.com/wasmCloud/wasmCloud/pull/2709/files#diff-87c740fda48fb006ffde1e67e879ab20a9b75aed0c38247f8edd4c7b5f7b9b5fR391-R419).

I think what should happen here is that we should:
1. move `io` future as-is (without awaiting it in any way or spawning a Tokio task) into the table
2. update `incoming_value_consume_sync` to `await` `io`, basically move the L102 from existing implementation there
3. update `incoming_value_consume_async` to return a new stream type (`HostInputStream` implementation https://docs.rs/wasmtime-wasi/latest/wasmtime_wasi/trait.HostInputStream.html), which in itself would:
    - hold the `io` future
    - use [`tokio::select!`](https://docs.rs/tokio/latest/tokio/macro.select.html) in [`Subscribe::ready`](https://docs.rs/wasmtime-wasi/latest/wasmtime_wasi/trait.Subscribe.html#tymethod.ready) implementation to poll I/O and check for more data

Reply:
Actually, I was wrong in the original message and there's no race condition here, the Tokio task may get leaked, however, it would be nice to use a `JoinSet` and store that in the table if we go with the proposed approach, this way at least the task would be dropped once the component instance is dropped